### PR TITLE
Note page creation and object versioning

### PR DESCRIPTION
**Pull request by krisx303**

### CODE REVIEW

File: src/main/java/com/brightpath/learnify/controller/NotesController.java [hunk 1]
```diff
@@ -111,4 +112,16 @@ public ResponseEntity<List<NoteSummaryDto>> listNotes(UUID workspaceId) {
                 .map(dtoMapper::asNoteSummaryDto)
                 .toList());
     }
+
+    @Override
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/NotesController.java:118]: We do not want to return this creatated page instead?

File: src/main/java/com/brightpath/learnify/controller/NotesController.java [hunk 2]
```diff
@@ -111,4 +112,16 @@ public ResponseEntity<List<NoteSummaryDto>> listNotes(UUID workspaceId) {
...
+    public ResponseEntity<String> createBoardNotePage(UUID noteId) {
+        notesService.createBoardNotePage(noteId);
+        return ResponseEntity.ok("Note page created");
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/NotesController.java]: and why ok but not created

File: src/main/java/com/brightpath/learnify/domain/note/NoteService.java [hunk 3]
```diff
@@ -79,22 +80,28 @@ public Note getNoteById(UUID uuid) {
...
+        if (rowsAffected == 0) {
+            throw new ResourceUpdateConflictException(BOARD_NOTE_PAGE, uuid);
+        }
     }
 
     @Transactional
-    public void updateDocumentNoteContentPage(UUID uuid, Integer pageNumber, String body) {
-        documentNotePageRepository.updateByNoteIdAndPageNumber(uuid, pageNumber, body);
+    public void updateDocumentNoteContentPage(UUID uuid, Integer pageNumber, String body, Integer version) {
+        int rowsAffected = documentNotePageRepository.updateByNoteIdAndPageNumber(uuid, pageNumber, body, version);
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/domain/note/NoteService.java:90]: how this rows affected works?

File: src/main/java/com/brightpath/learnify/domain/note/NoteService.java [hunk 4]
```diff
@@ -105,4 +112,24 @@ public List<Note> searchNotes(String userId, @Nullable UUID workspaceId) {
...
+        NoteEntity note = noteRepository.findById(noteId)
+                .orElseThrow(() -> new ResourceNotFoundException(NOTE));
+        int newPageNumber = note.getPagesCount() + 1;
+        boardNotePageRepository.save(new BoardNotePageEntity(uuidProvider.generateUuid(), noteId, newPageNumber, "", 1));
+        note.setPagesCount(newPageNumber);
+        note.setUpdatedAt(OffsetDateTime.now(Clock.systemUTC()));
+        noteRepository.save(note);
+    }
+
+    public void createDocumentNotePage(UUID noteId) {
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/domain/note/NoteService.java:124]: documents pages aren't automaticly created by just entering?

File: src/main/resources/api.yaml [hunk 5]
```diff
@@ -148,6 +143,25 @@ paths:
...
+  /notes/{noteId}/board/pages:
+    post:
+      summary: Create a new page in board note
+      operationId: createBoardNotePage
+      tags:
+        - notes
+      parameters:
+        - $ref: '#/components/parameters/noteId'
+      responses:
+          '201':
```
Comment (Adam3004) [inline src/main/resources/api.yaml:155]: thats why i ask about ResponseEntity.ok. We suggest it will be 201 created but return 200 ok

**Review by Adam3004 — APPROVED** [review]

File: src/main/java/com/brightpath/learnify/controller/NotesController.java (hunk 1 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/controller/NotesController.java:118]: We could, but I don't think it is necessary for now

File: src/main/java/com/brightpath/learnify/domain/note/NoteService.java (hunk 3 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/domain/note/NoteService.java:90]: The sql update query returns how many records were updated. Normally we expect that one row was affected - if not then we assume that provided version is not actual to allow the update so we throw Confilct exception

File: src/main/java/com/brightpath/learnify/controller/NotesController.java (hunk 2 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/controller/NotesController.java]: Right, it should be 201

File: src/main/java/com/brightpath/learnify/controller/NotesController.java (hunk 1 above)
Comment (Adam3004) [reply src/main/java/com/brightpath/learnify/controller/NotesController.java:118]: so when it will be?

Comment (krisx303) [reply src/main/java/com/brightpath/learnify/controller/NotesController.java:118]: I would say that if thats not required now it won't be required at all 😅
We can extend the endpoint in the future for creating pages in the midde - then we could return the created page number

**Pull request merged**